src/DotNetCore.CAP.AzureServiceBus/CAP.AzureServiceBusOptions.cs

```diff
@@ -62,13 +65,22 @@ public class AzureServiceBusOptions
         /// </summary>
         public Func<ServiceBusReceivedMessage, List<KeyValuePair<string, string>>>? CustomHeaders { get; set; }
 
+        /// <summary>
+        /// Use this function to write additional headers from the original ASB Message or any Custom Header, i.e. to allow compatibility with heterogeneous systems, into <see cref="CapHeader"/>       
+        /// </summary>
+        /// <remarks>
+        /// This property is a copy cat of <see cref="CustomHeaders"/>. The difference is: It also accpets a <see cref="IServiceProvider"/> to use DI services.
+        /// This is added due to converting <see cref="SnowflakeId"/> into a singleton service.
+        /// </remarks>
+        public Func<ServiceBusReceivedMessage, IServiceProvider, List<KeyValuePair<string, string>>>? CustomHeadersBuilder { get; set; }
```

Comment:
I suggest Add `[Obsolete("xxxxx")]` for old `CustomHeaders` option, we will  remove old option at the feature